src/WebJobs.Extensions.DurableTask/Listener/DurableTaskListener.cs

```diff
@@ -30,8 +47,18 @@ internal sealed class DurableTaskListener : IListener
                 throw new ArgumentNullException(nameof(functionName));
             }
 
+            this.functionId = functionId;
             this.functionName = functionName;
             this.functionType = functionType;
+            this.storageConnectionString = storageConnectionString;
+#if !FUNCTIONS_V1
+            this.scaleMonitor = new Lazy<DurableTaskScaleMonitor>(() => new DurableTaskScaleMonitor(
```

Comment:
Make sure to handle the case where `storageConnectionString` is null, (i.e. when a customer is using a non AzureStorage provider).

Reply:
I believe that case is actually covered, but would love a second opinion. StorageConnectionString only needs to be valid if they've enabled runtime scaling. The scale monitor will be created no matter what, but the `DisconnectedPerformanceMonitor` won't be created unless runtime scaling is enabled -- at which point it will throw an exception if the connection string is null. So a customer that's using non-azure storage won't be affected unless they enable runtime scaling.